components/server/src/workspace/workspace-factory.ts

```diff
@@ -135,7 +136,10 @@ export class WorkspaceFactory {
         const span = TraceContext.startSpan("createForCommit", ctx);
 
         try {
-            const config = await this.configProvider.fetchConfig({span}, user, context);
+            const [ config, project ] = await Promise.all([
+                this.configProvider.fetchConfig({ span }, user, context),
+                this.projectDB.findProjectByCloneUrl(context.repository.cloneUrl),
```

Comment:
This reads like a project owns a repository exclusively. Is that meant to be? This feels like it's going to be a 100% shadow of a GitHub org, and I thought this would not cover all our usecases... but maybe have some reading to do, here.

Reply:
Yes, that is how it works. Having multiple projects per repo isn't really supported by GitHub, because I can install the GitHub app only once per org. But also the use cases I've heard so far have been are rather edge cases.

Reply:
> This reads like a project owns a repository exclusively.

A team owns a projects, yes.

A repository is not meant to be used in multiple projects. The `cloneUrl` should be unique.

> this would not cover all our usecases

What do we miss?

Reply:
See comment above

> What do we miss?

Before I thought about a "Team" as a way to have my custom "config overlay" over arbitrary projects, which are _my_ representation of a GH repo. But I understand that this is not easily doable with GH because of the "write access constraint" necessary for GH Apps, and might be way to complicated for most of our usecases. :+1: